src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Utilities/TypeStyle/CSharpUseImplicitTypeHelper.cs

```diff
@@ -225,7 +211,48 @@ protected override bool IsStylePreferred(in State state)
             var newDeclarationTypeNode = newTree.GetRoot(cancellationToken).GetAnnotatedNodes(annotation).Single();
             var newDeclarationType = newSemanticModel.GetTypeInfo(newDeclarationTypeNode, cancellationToken).Type;
 
-            return SymbolEquivalenceComparer.Instance.Equals(declarationType, newDeclarationType);
+            return SymbolEquivalenceComparer.TupleNamesMustMatchInstance.Equals(
+                declarationType, newDeclarationType);
+        }
+
+        private bool IsMatchingArgToSimpleMethod(DeclarationExpressionSyntax declarationExpression, SemanticModel semanticModel, CancellationToken cancellationToken)
+        {
+            if (!(declarationExpression.Parent is ArgumentSyntax argument) ||
+                !(argument.Parent is ArgumentListSyntax argumentList) ||
+                !(argumentList.Parent is InvocationExpressionSyntax invocationExpression))
+                return false;
+
+            // If there was only one member in the group, and it was non-generic itself,
+            // then this change is safe to make without doing any complex analysis.
```

Comment:
Seems like this comment needs a rewrite now. Since the rest of this method is proving that this comment was wrong...

Reply:
done